include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -58,6 +66,11 @@ extern "C" {
 #define NO_EXCEPTION
 #endif
 
+typedef struct OrtDeleter {
+  void(ORT_API_CALL* f)(void* param) NO_EXCEPTION;
```

Comment:
Do we need a struct in the API or could we use an opaque type like we do for all the stuff around line 147?